hashintel/hash
# H-1077, H-1129: Fix file uploads, deduplicate logic

## 🌟 What is the purpose of this PR?

File uploads were broken with the introduction of the permission system. This was because we were attempting to make an entity public that was already public, which threw an error which stopped the file upload process.

In fixing this I've:
1. Updated all org/user avatar/cover image functions to use the central file uploads context to save having to fix the same issue in multiple places
2. Fixed the local file storage upload, which needed updating for moving from a multi-part form upload to just sending the file data on its own in the request

## Pre-Merge Checklist 🚀

### 🚢 Has this modified a publishable library?




This PR:


### 📜 Does this require a change to the docs?




The changes in this PR:



### 🕸️ Does this require a change to the Turbo Graph?




The changes in this PR:


## ❓ How to test this?



1. Try adding / updating an avatar from an org's setting page, or a user or org's profile page
2. See if other users can view it (who aren't members of the same org)
